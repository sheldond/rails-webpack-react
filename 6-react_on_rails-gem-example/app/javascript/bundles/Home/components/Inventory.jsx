import React from "react"
import PropTypes from "prop-types"

class Inventory extends React.Component {
  constructor() {
    super();
    this.renderInventory = this.renderInventory.bind(this);
  }

  renderInventory(key) {
    const fish = this.props.fishes[key];
    return (
      <div className="fish-edit" key={key}>
        <input type="text" value={fish.name} name="name" placeholder="Fish Name" onChange={(event) => this.handleChange(event, key)}/>
        <input type="text" value={fish.price} name="price" placeholder="Fish Price" onChange={(event) => this.handleChange(event, key)} />
        <select type="text" value={fish.status} name="status" placeholder="Fish Status" onChange={(event) => this.handleChange(event, key)}>
          <option value="available">Fresh!</option>
          <option value="unavailable">Sold Out!</option>
        </select>
        <textarea type="text" value={fish.desc} name="desc" placeholder="Fish Desc" onChange={(event) => this.handleChange(event, key)}></textarea>
        <input type="text" value={fish.image} name="image" placeholder="Fish Image" onChange={(event) => this.handleChange(event, key)} />
        <button onClick={() => this.props.removeFish(key)}>Remove Fish</button>
      </div>
    )
  }

  render() {
    return (
      <div>
        <h2>Fish Inventory</h2>
        { Object.keys(this.props.fishes).map(this.renderInventory) }
      </div>
    )
  }
}

Inventory.propTypes = {
  fishes: PropTypes.object
};
export default Inventory
