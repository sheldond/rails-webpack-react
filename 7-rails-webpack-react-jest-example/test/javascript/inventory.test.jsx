import React from 'react';
import ReactDOM from 'react-dom';
import Inventory from '../../app/javascript/inventory';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

const fishes = {
  fish1: {
    name: 'Pacific Halibut',
    image: 'https://i.istockimg.com/file_thumbview_approve/36248396/5/stock-photo-36248396-blackened-cajun-sea-bass.jpg',
    desc: 'Everyones favorite white fish. We will cut it to the size you need and ship it.',
    price: 1724,
    status: 'available'
  }
}

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Inventory fishes={fishes} />, div);
});

test('matches the rendered snapshot', () => {
  const component = renderer.create(
    <Inventory fishes={fishes} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('fish name is correct', () => {
  const inventory = Enzyme.shallow(
    <Inventory fishes={fishes} />
  );
  expect(inventory.find("input[name='name']")
    .prop('value')).toEqual('Pacific Halibut')
});
