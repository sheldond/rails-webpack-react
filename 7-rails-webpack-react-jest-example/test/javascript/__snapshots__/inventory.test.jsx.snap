// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the rendered snapshot 1`] = `
<div>
  <h2>
    Fish Inventory
  </h2>
  <div
    className="fish-edit"
  >
    <input
      name="name"
      onChange={[Function]}
      placeholder="Fish Name"
      type="text"
      value="Pacific Halibut"
    />
    <input
      name="price"
      onChange={[Function]}
      placeholder="Fish Price"
      type="text"
      value={1724}
    />
    <select
      name="status"
      onChange={[Function]}
      placeholder="Fish Status"
      type="text"
      value="available"
    >
      <option
        value="available"
      >
        Fresh!
      </option>
      <option
        value="unavailable"
      >
        Sold Out!
      </option>
    </select>
    <textarea
      name="desc"
      onChange={[Function]}
      placeholder="Fish Desc"
      type="text"
      value="Everyones favorite white fish. We will cut it to the size you need and ship it."
    />
    <input
      name="image"
      onChange={[Function]}
      placeholder="Fish Image"
      type="text"
      value="https://i.istockimg.com/file_thumbview_approve/36248396/5/stock-photo-36248396-blackened-cajun-sea-bass.jpg"
    />
  </div>
</div>
`;
